apis/v1alpha1/tlsroute_types.go

```diff
@@ -0,0 +1,131 @@
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// TLSRouteSpec defines the desired state of TLSRoute
+type TLSRouteSpec struct {
+	// Rules are a list of TLS matchers and actions.
+	// +required
+	// +kubebuilder:validation:MinItems=1
+	Rules []TLSRouteRule `json:"rules" protobuf:"bytes,1,rep,name=rules"`
```

Comment:
Did you consider following the HTTPRoute Host/Rules pattern here?

https://github.com/kubernetes-sigs/service-apis/pull/238#issuecomment-666008041

Reply:
I did consider it but decided against it because: 

> Disadvantages are that the binding to the SNI requirement is less obvious, which could make the API harder to reason about and prone to misunderstanding. Maybe

Reply:
Ok, I can sort of see that, but I realized that we no longer have anything that we can have a virtual host concept on in the TCP or TLS routes. That's awkward because a virtualhost seems like a natural boundary to want to attach an extension (e.g. this host needs some custom protocol on top of UDP).

Anyway, we don't need to hold up this PR discussing this problem :)